lib/utils.js

```diff
@@ -5,58 +5,65 @@ module.exports = {
         return crypto.createHash("md5").update(str, "utf8").digest("hex")
     },
 
-    checkUniqueHash: function(items, source, properties) {
+    addUniqueHashedObject: function(items, source, properties) {
```

Comment:
if you're going to add it, you might as well add doc.

Reply:
That moment when nothing else has doc. 😿 

Will do, though.

Reply:
I'm starting to feel like this method is doing too much stuff, checkUniqueHash was one thing but now we have two things? Why even do this refactor you don't remove much code and I feel like this makes things more complicated/obstructed/magical?